### PR TITLE
Master component listings on marketplace

### LINKED ISSUE #264: Server user sells infinite components at a fixed price.
**Describe the solution you'd like**
Make a server user that sells an infinite amount of components at a fixed price.

**Describe alternatives you've considered**
Make a server user that has an infinite listing, and users can request a specific amount from the listing.

**Additional context**
Also, the CSS on the server user listing needs to be different.

### PR DESCRIPTION
**Pull request by Derplime**
**Description**
Modified the Transaction API for Components (and Casus Blocks :[ ) so that if the seller is the master account, the listing isn't deleted. User created listings are still removed like normal.

Few key things in this PR:
* For what it's worth, I can't really make an infinite amount of items for the master account to sell, but I can make it really large. So I gave the master account 2 billion of each component. I doubt we'll be able to effectively run this dry.
* I edited the `.env` file for this. Be sure to update yours from the drive.
* The DB backups were updated as of this PR, so update those backups if you downloaded them before this PR went came out. Changes include:
  * Changed master account component amount to 2 billion per
  * Added user default tanks to non-master accounts
    * All non-master account ObjectIds have changed because of this.
  * Added DB backup of master Marketplace component sales.
* Lastly, added a small edit to have the new user starter tanks contain their name (prevent swarm of tanks initially named "Starter Tank")

Resolves #264 

**Type of change**
Check options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Fix or feature that would cause existing functionality to not work as expected
- [ ] This change requires a update in the design document

**Steps to Verify Functionality**
1. Get updated `.env` and DB backups and import them.
2. Login as master account
3. Make a component sale of any component
4. Log out and login as a non-master user.
5. Buy the component listed by the master account.
6. Confirm it isn't deleted.
(Optional to confirm the non-master sales logic still works)
7. List a component sale as the non-master user.
8. Log out and login as another user.
9. Buy the listed component sale.
10. Confirm it gets deleted.

**Final Checklist:**
Go down the list and check them off.

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the design document (if applicable)
- [x] My changes generate no new warnings
- [x] Pulled updated master branch (if changed since branch creation) and corrected any conflicts, if any occur.

### CODE REVIEW

File: src/backend/controllers/marketController.js [hunk 1]
```diff
@@ -442,15 +445,20 @@ exports.marketTransaction = async (req: Request, res: Response) => {
...
+                    await MarketSale.deleteOne({ _id: saleId }, (err: Error) => {
+                        if (err) {
+                            console.error(err.message);
+                            return res
+                                .status(500)
+                                .json({ msg: 'Could not delete market sale.' });
+                        }
+                    });
+                } else {
+                    console.log('Sale belongs to master account. Not deleting.')
```
Comment (Derplime) [inline src/backend/controllers/marketController.js:460]: I had these here for notification purposes between user and master listings, but let me know if I should remove them.

File: src/backend/controllers/marketController.js [hunk 2]
```diff
@@ -490,15 +498,20 @@ exports.marketTransaction = async (req: Request, res: Response) => {
...
```
Comment (Derplime) [inline src/backend/controllers/marketController.js:513]: ditto for this

File: src/backend/controllers/marketController.js (hunk 1 above)
Comment (SecondThread) [reply src/backend/controllers/marketController.js:460]: That should be fine. As long as it isn't an oversight, I'm cool with it

File: src/backend/controllers/userController.js [hunk 3]
```diff
@@ -78,7 +78,7 @@ exports.register = async (req: Request, res: Response) => {
 		// Create initial tank for user
 		let tank = new Tank();
 		tank.userId = user.id;
-		tank.tankName = 'Starter Tank';
+		tank.tankName = `${userName}'s Starter Tank`;
```
Comment (SecondThread) [inline src/backend/controllers/userController.js:81]: Are users allowed to rename tanks they have already created?

**Review by SecondThread — APPROVED** [review]
The code looks good to me

File: src/backend/controllers/userController.js (hunk 3 above)
Comment (Derplime) [reply src/backend/controllers/userController.js:81]: It's outside the scope of this PR, but the API to update the tank supports it

**Pull request merged**